create and read .resource file

milljm opened the pull request:
Refs #20729

Commit:
WIP: create and read .resource file

Refs #20729

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/1038913/) on 8509f91 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/20968/site)

This comment will be updated on new commits.

Commit:
Support older Python versions

Commit:
Do not error on failed . resource file creation

The TestHarness will 'handle' these situations (skip that test, with
a helpful caveat)

update our .gitignore list to include .moose_test resource file.

moosebuild commented:
Job [Min clang](https://civet.inl.gov/job/1039372/) on 88a867a : invalidated by @milljm

updated centos-7 container to include py-yaml
